app/routes.py

```diff
@@ -0,0 +1,177 @@
+    return response
+
+
+@app.route('/shorten', methods=['POST'])
+def create_shortlink():
+    """
+    * Quick summary of the function *
+
+    The create_shortlink route's goal is to take an url and create a shortlink to it.
+    The only parameter we should receive is the url to be shortened, within the post payload.
+
+    We extract the url to shorten, the request scheme, domain and base path and send them to a
+    function to check those parameters. (check_params)
+
+    If those parameters are checked, we create the response to be sent to the user and create the
+    shortlink to send back (add_item)
+
+    * Abortions originating in this function *
+
+    Abort with a 403 status code if the Origin header is not set nor one we expect.
+
+    * Abortions originating in functions called from this function *
+
+    Abort with a 400 status code from check_params, add_item
+    Abort with a 500 status code from add_item
+
+    * Parameters and return values *
+
+    :request: the request must contain a Origin Header, and a json payload with an url field
+    :return: a json in response which contains the url which will redirect to the initial url
+    """
+    logger.info("Shortlink Creation route entered at %f", time.time())
+    if request.headers.get('Origin') is None or not \
+            re.match(Config.allowed_domains_pattern, request.headers['Origin']):
+        logger.critical("Shortlink Error: Invalid Origin")
+        abort(make_error_msg(403, "Not Allowed"))
+    response_headers = base_response_headers
+    try:
+        url = request.json.get('url', None)
+    except BadRequest:
+        logger.error("No Json Received as parameter")
+        abort(make_error_msg(400, "This service requires a json to be posted as a payload."))
+    scheme = request.scheme
+    domain = request.url_root.replace(
+        scheme, ''
+    )  # this will return the root url without the scheme
+    base_path = request.script_root
+    logger.debug(
+        "params received are : url: %s, scheme: %s, domain: %s, base_path: %s",
+        url, scheme, domain, base_path
+    )
+    base_response_url = check_params(scheme, domain, url, base_path)
+    table = get_dynamodb_table()
+    response = make_response(
+        jsonify({
+            "shorturl": ''.join(base_response_url + add_item(table, url)), 'success': True
+        })
+    )
+    response.headers = response_headers
+    response_headers['Access-Control-Allow-Origin'] = request.headers['origin']
+    response_headers['Access-Control-Allow-Methods'] = 'POST, OPTION'
+    response_headers['Access-Control-Allow-Headers'] = 'Content-Type, Authorization,' \
+                                                       ' x-requested-with, Origin, Accept'
+    logger.info(
+        "Shortlink Creation Successful. Returning the following response: %s", str(response)
+    )
+    return response
+
+
+@app.route('/redirect/<url_id>', methods=['GET'])
+def redirect_shortlink(url_id):
```

Comment:
Changing the route name is not a bad idea, as the current names and separation are a legacy from chsdi. It makes sense to have 
- One endpoint to create resources
- One endpoint to pull resources (with an option to directly use it)

For the parameter redirect, what would be the basic use-case ? 
If we go with how most shortlinks are done in the world, the base use case is to redirect by default. It would make more sense to have the ```/shortlinks/<shortlink_id>``` be the redirect, and have ```shortlink/<id>?redirect=false``` to be the json endpoint.

Reply:
For now, I've implemented it with one endpoint for both redirection and information fetching, with the default comportment being to redirect if there is no redirect parameter set to false. 

Changing this to the other comportment is a matter of changing one string in the code, and changing one test (the one where I explicitly provide the parameter for the default case) to its counterpart, which should be a 2-3 minutes task at most.

Reply:
The default REST behavior is that you get back the resource from an endpoint, in the case of the shortlink service this is the 'long' url. So I would suggest to have the default being the standard REST behavior, whereas the redirection has to be explicitly asked for.

Reply:
Works for me. I'll change this after lunch and we should be good to merge.

Reply:
It's been done.